src/utils.rs

```
@@ -0,0 +1,8 @@
+pub fn get_struct_name<T>() -> String {
+    std::any::type_name::<T>()
+        .split("::")
+        .collect::<Vec<&str>>()
+        .get(2)
+        .unwrap()
```

We should avoid use of `unwrap()` since it is unsafe. Rather we should propagate the error and treat this as a `Result` if it can err.

---

TODO